### ISSUE TITLE
Screenshot bug

### ISSUE THREAD

**Issue by josejl1987** [Rinnegatamante/rinCheat#16]
I tried the screenshot function on Trails of Cold Steel II,and the resulting image is a bit off.
I think it's just that the red channel on the bmp output is one pixel off to the left.
I made a manual edit on GIMP so you can see the difference.

![Original screenshot](https://cloud.githubusercontent.com/assets/3105365/18446536/05d3d414-7923-11e6-930e-fe3605efa263.png)
![Manual edit](https://cloud.githubusercontent.com/assets/3105365/18446538/05f5ebb2-7923-11e6-851b-6c4e79e03e20.png)

**Comment by Rinnegatamante** [comment]
Screenshots look the same to me.

PS: I suppose the game runs in RAM mode.

**Comment by gnmmarechal** [comment]
They don't. Look at the left of the top screenshot. You see the red line. also, it looks blurrier than the bottom one?

**Comment by josejl1987** [comment]
Yeah,it runs on RAM mode. 

The difference is slightly noticeable ,but it's there.

**Comment by Rinnegatamante** [comment]
Yep noticed right now but i don't know why this happens. On next release MMC screenshots will be fixed. I suggest you to wait that release andt hen we can check if the problem appears in MMC too.

**Comment by Rinnegatamante** [comment]
@josejl1987 Can you check if using this solves the problem?
[rinCheat.zip](https://github.com/Rinnegatamante/rinCheat/files/468301/rinCheat.zip)

**Comment by josejl1987** [comment]
Still the same, I'm afraid. 

I took a quick glance at the screenshot code, and I don't see anything strange myself.

What color format does the Vita framebuffer use?  From your code I'd say it's B8G8R8A8. 

Is there any chance it could be using some strange 16 bit mode?  Falcom games did it on PSP,and sometimes I see color banding similar to 16 bit images.

Would it be hard to dump the framebuffer in its original format?

**Comment by Rinnegatamante** [comment]
It uses ABGR: https://github.com/vitasdk/vita-headers/blob/master/include/psp2/display.h#L34

**Comment by DarkPulse** [comment]
I can confirm this also happens on screenshots from Dragon Quest Builders.

Original:
![pcse00912_1024_017_original](https://cloud.githubusercontent.com/assets/5454684/19825689/6aeb68ee-9d4a-11e6-8299-dee17aa6cd13.png)

Fixed:
![pcse00912_1024_017_fixed](https://cloud.githubusercontent.com/assets/5454684/19825695/785ac984-9d4a-11e6-95a1-70987d309536.png)

**Comment by Rinnegatamante** [comment]
IF you really need good screenshots, atm you can start the game by holding L trigger. This will start the game in MMC mode where screenshots should be fixed.

I'm still investigating the issue to find a solution.
